### PR TITLE
[13.0][MIG] migration of module sale_cancel_reason

### PR DESCRIPTION
**Pull request by alan196**
### Proposed changes

I've migrated the module sale_cancel_reason and its functionalities from Odoo 12.0 to 13.0. I've checked if all unit tests are still running.

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Migration Update

### Checklist

- [x] I have read the CONTRIBUTING doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### Further comments

There is an existing PR https://github.com/OCA/sale-workflow/pull/945 to migrate this module, but it has errors and doesn't have activity for a year.

### CODE REVIEW

**Review by jesus01x — APPROVED** [review]

File: sale_cancel_reason/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+To use this module, you need to:
+
+* Click at "Cancel Order" button from a sales order wich state equal
```
Comment (bodedra) [inline sale_cancel_reason/readme/USAGE.rst]:
```suggestion
* Click at "Cancel Order" button from a sales order which state equal
```

**Review by bodedra — COMMENTED** [review]
I have reviewed in runbot and it doesn't stamp reason in sale order.

Screenshot from 2020-11-02 08-46-57

**Comment by alan196** [comment]
@bodedra Changes applied, could you help me to review again 🙏

**Comment by bodedra** [comment]
Function review. :+1: 

Would you please fix this?

Screenshot from 2020-11-04 07-57-07

**Review by bodedra — APPROVED** [review]
Function review. :+1:

**Comment by bodedra** [comment]
> Function review. +1
> 
> Would you please fix this?
> 
> Screenshot from 2020-11-04 07-57-07

Answered my self. Here is fix https://github.com/OCA/sale-workflow/pull/1313

I have restarted CI.

**Comment by bodedra** [comment]
Suggestion:

IMO it should first stamp "Cancelled" status and then "Reason". Currently it shows "Reason" first.

Screenshot from 2020-11-04 08-24-40

**Comment by alan196** [comment]
> Suggestion:
> 
> IMO it should first stamp "Cancelled" status and then "Reason". Currently it shows "Reason" first.
> 
> Screenshot from 2020-11-04 08-24-40

This is Odoo native functionality, this cannot be controlled using the default `tracking` attribute in the field. It may need to modify the cancel method.

**Comment by rousseldenis** [comment]
> > Suggestion:
> > IMO it should first stamp "Cancelled" status and then "Reason". Currently it shows "Reason" first.
> > Screenshot from 2020-11-04 08-24-40
> 
> This is Odoo native functionality, this cannot be controlled using the default `tracking` attribute in the field. It may need to modify the cancel method.

Indeed. But maybe it could be a solution. It would avoid overriding cancel button in form view. Launching the wizard from the cancel function could be a good lead.

**Comment by bodedra** [comment]
Instead of `tracking` attribute, we can use `message_post()`.

**Review by tonygalmiche — APPROVED** [review]
LGTM : Functional review only with runbot

image

**Comment by newtratip** [comment]
:+1: Functional Test

**Review by newtratip — APPROVED** [review]

**Comment by alan196** [comment]
@rousseldenis this PR is ready to merge, could you help me to merge

Thanks in advance

**Comment by kittiu** [comment]
Hello, I do have requirement to do cancel reason like this for many document. I.e., SO, PR, PO, EX, INV. Instead of doing it one by one,
I am thinking, from v14, I will make abstract base_cancel_reason + sale_cancel_reason to start with.

Any opinion?

**Pull request merged**